keyboards/shahnetwork/b/ld/STM32F412xE_EEPROM.ld

```diff
@@ -0,0 +1,85 @@
+/*
+    ChibiOS - Copyright (C) 2006..2018 Giovanni Di Sirio
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+
+/*
+ * STM32F412xE memory setup.
+ */
+MEMORY
+{
+    flash0 (rx) : org = 0x08000000, len = 16k
+    flash1 (rx) : org = 0x08004000, len = 64k
+    flash2 (rx) : org = 0x08010000, len = 512k - 80k
```

Comment:
Because the chip has only 4 16K flash sectors, the memory setup should be like this if you intend to use 48K for the backing store:
```suggestion
    flash0 (rx) : org = 0x08000000, len = 16k
    flash1 (rx) : org = 0x08004000, len = 48k
    flash2 (rx) : org = 0x08010000, len = 512k - 64k
```
(Another option is to use just 16K, like for the STM32F4x1 chips.)